### PR TITLE
Enhance Codebase with Dynamic Imports and Babel Aliases

### PR DESCRIPTION
**Pull request by purevsurena**
This pull request introduces improvements to the React Native app by configuring Babel aliases for enhanced module resolution. In three isolated commits, the codebase has been refactored to replace static imports with dynamic ones, providing greater flexibility and maintainability.

Tasks:

- [x] Configure Babel Alias for Improved Module Resolution:
- [x] Replaced static imports with dynamic ones in the App.tsx
- [x] Replaced static imports with dynamic ones in the main-navigator.tsx
- [x] Replaced static imports with dynamic ones in all components
- [x]  Replaced static imports with dynamic ones on all screens
- [x]  Replaced static imports with dynamic ones in the utils and helpers

Screenshot:
Screenshot 2023-11-01 at 00 17 37

I recommend granting permission to proceed with the secure replacement of all static imports in the codebase. This action is poised to optimize the application's code structure, fostering a more resilient foundation for future development. If agreeable, I am prepared to take on this task.

### CODE REVIEW

**Comment by purevsurena** [comment]
Hello @belgamo , Please review the above PR.

**Comment by belgamo** [comment]
Thanks for the contribution! Please, you can use this PR and go ahead to do the replacements for the other files.

**Review by belgamo — APPROVED** [review]

**Comment by purevsurena** [comment]
> Thanks for the contribution! Please, you can use this PR and go ahead to do the replacements for the other files.

Thank you for your comment. I've replaced all remaining code and conducted a local check, including the full flow. You can now proceed to merge it.

By the way, please ensure to clear the cache and start Metro for the first time.

File: apps/mobile/src/components/register-adoption-form/adoption-form.component.tsx [hunk 1]
```diff
@@ -6,13 +6,13 @@ import React from 'react';
 import { FormProvider, useForm } from 'react-hook-form';
 import { Platform } from 'react-native';
 
-import { AdoptionSteps } from './adoption-form.types';
-import FormSteps from './compose/form-steps';
-import StepperController from './compose/stepper-controller';
-import StepperIndicator from './compose/stepper-indicator';
-import { useMultiStepNavigation } from './hooks/use-multi-step-navigation.hook';
```
Comment (belgamo) [inline apps/mobile/src/components/register-adoption-form/adoption-form.component.tsx]: All these four imports are actually part of the `register-adoption-form` component. They aren't meant to be used outside of their host component, that's the purpose of the `compose` folder. WDYT about keeping the dot notation import in this case?

File: apps/mobile/src/components/tabs/tabs.component.tsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import React, { useState } from 'react';
 import { useWindowDimensions } from 'react-native';
 import { TabView, SceneMap } from 'react-native-tab-view';
 
-import CustomTabBar from './compose/tab-bar.component';
```
Comment (belgamo) [inline apps/mobile/src/components/tabs/tabs.component.tsx]: Same

File: apps/mobile/src/screens/get-location/get-location.screen.tsx [hunk 3]
```diff
@@ -5,15 +5,16 @@ import { useEffect } from 'react';
...
-import Routes from '../../routes';
-import theme from '../../theme';
+
+import localizationImg from '@/assets/localization.png';
+import SafeArea from '@/components/safe-area/safe-area';
+import AppStatusBar from '@/components/status-bar/status-bar.component';
+import { useAuth } from '@/hooks/use-auth-provider';
+import useLocale from '@/hooks/use-locale';
+import useGeolocation from '@/hooks/use-user-location';
+import { Warnings } from '@/hooks/use-user-location/use-user-location';
```
Comment (belgamo) [inline apps/mobile/src/screens/get-location/get-location.screen.tsx]: Can you reexport `Warnings` from the index file so we can import both modules at the same line?

**Review by belgamo — CHANGES_REQUESTED** [review]
A few more changes before merging 😄

**Comment by purevsurena** [comment]
Hey @belgamo ,

I've addressed the import issue related to the compose folder as suggested. The warning has been resolved and the import now fits into a single line.

I believe the changes are ready for merging. Feel free to review.

**Review by belgamo — APPROVED** [review]

**Pull request merged**